lib/olive_branch_patch.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# This will monkey patch olivebranch middleware https://github.com/vigetlabs/olive_branch/blob/master/lib/olive_branch/middleware.rb
+# so that when the VA inflection changes a key like something_va_something, the results will
+# have the properly inflected somethingVASomething as well as support the old somethingVaSomething.
+# This is a deprecation path and should be removed once consumers have adopted the inflection
+
+module OliveBranchMiddlewareExtension
+  def call(env)
+    result = super(env)
+    _status, _headers, response = result
+    if env['HTTP_X_KEY_INFLECTION'] =~ /camel/i
+      response.each do |json|
+        # do not process strings that aren't json (like pdf responses)
+        next unless json.is_a?(String) && json.starts_with?('{')
+
+        duplicate_basic_va_keys!(json)
+        duplicate_collection_va_keys!(json)
+      end
+    end
+    result
+  end
+
+  private
+
+  VA_KEY_VALUE_PAIR_REGEX = /("[^"]+VA[^"]*"):("[^"]*"|\d+|true|false|null)/.freeze
+
+  def duplicate_basic_va_keys!(json)
+    json.gsub!(VA_KEY_VALUE_PAIR_REGEX) do
```

Comment:
this method changed from the original version of the patch https://github.com/department-of-veterans-affairs/vets-api/pull/5133/files
now using the regex references instead of trying to split on `:`